packages/subgraph/src/utils/tokens/erc20.ts

```diff
@@ -95,6 +95,31 @@ export function fetchERC20(address: Address): ERC20Contract | null {
   return contract;
 }
 
+/**
+ * @dev Determines the type of ERC20 token (wrapped or regular) and returns its address.
+ *
+ * @param token The address of the token to be determined.
+ * @return tokenAddress The address of the ERC20 token if it's either wrapped or regular, null otherwise.
+ */
+export function determineERC20Token(token: Address): string | null {
+  let tokenAddress: string;
+  if (supportsERC20Wrapped(token)) {
+    let contract = fetchWrappedERC20(token);
+    if (!contract) {
+      return null;
+    }
+    tokenAddress = contract.id;
```

Comment:
Unrelated to this PR, but should token have maybe a separate `address` field that is then also returned as the ID? Seems cleaner to me.
Ignore this comment if this is done on purpose for performance reasons.